src/libs/DistanceRequestUtils.ts

```diff
@@ -122,14 +185,50 @@ function getDistanceMerchant(
         return translate('iou.routePending');
     }
 
-    const formattedDistance = getDistanceForDisplay(hasRoute, distanceInMeters, unit, rate, translate);
-    const singularDistanceUnit = unit === CONST.CUSTOM_UNITS.DISTANCE_UNIT_MILES ? translate('common.mile') : translate('common.kilometer');
-    const ratePerUnit = PolicyUtils.getUnitRateValue(toLocaleDigit, {rate});
+    const distanceInUnits = getDistanceForDisplay(hasRoute, distanceInMeters, unit, rate, translate);
+    const ratePerUnit = getRateForDisplay(unit, rate, currency, translate, toLocaleDigit);
 
-    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-    const currencySymbol = CurrencyUtils.getCurrencySymbol(currency) || `${currency} `;
+    return `${distanceInUnits} @ ${ratePerUnit}`;
+}
+
+/**
+ * Retrieves the mileage rates for given policy.
+ *
+ * @param policy - The policy from which to extract the mileage rates.
+ *
+ * @returns An array of mileage rates or an empty array if not found.
+ */
+function getMileageRates(policy: OnyxEntry<Policy>): Record<string, MileageRate> {
+    const mileageRates: Record<string, MileageRate> = {};
+
+    if (!policy) {
+        return mileageRates;
+    }
+
+    if (!policy?.customUnits) {
+        return mileageRates;
+    }
+
+    const distanceUnit = Object.values(policy.customUnits).find((unit) => unit.name === CONST.CUSTOM_UNITS.NAME_DISTANCE);
+    if (!distanceUnit?.rates) {
+        return mileageRates;
+    }
+
+    Object.entries(distanceUnit.rates).forEach(([rateID, rate]) => {
+        mileageRates[rateID] = {
+            rate: rate.rate,
+            currency: rate.currency,
+            unit: distanceUnit.attributes.unit,
+            name: rate.name,
+            customUnitRateID: rate.customUnitRateID,
+        };
+    });
+
+    return mileageRates;
+}
 
-    return `${formattedDistance} @ ${currencySymbol}${ratePerUnit} / ${singularDistanceUnit}`;
+function getRateForP2P(currency: string): RateAndUnit {
```

Comment:
NAB: Noting that this is where we need to handle getting the rate from the transaction `defaultP2PRate`

Reply:
@neil-marcellini you mean adding a comment what this function does?

Reply:
Implementing [this](https://github.com/Expensify/App/pull/38543#discussion_r1572686251)